src/Toml/PrefixTree.hs

```diff
@@ -177,9 +182,24 @@ lookupT lk (Branch pref mv prefMap) =
 lookup :: Key -> PrefixMap a -> Maybe a
 lookup k@(p :|| _) prefMap = HashMap.lookup p prefMap >>= lookupT k
 
--- | Constructs 'PrettyMap' structure from the given list of 'Key' and value pairs.
+-- | Constructs 'PrefixMap' structure from the given list of 'Key' and value pairs.
 fromList :: [(Key, a)] -> PrefixMap a
 fromList = foldl' insertPair mempty
   where
     insertPair :: PrefixMap a -> (Key, a) -> PrefixMap a
     insertPair prefMap (k, v) = insert k v prefMap
+
+-- | Converts 'PrefixTree' to the list of pairs.
+toListT :: PrefixTree a -> [(Key, a)]
+toListT (Leaf k v) = [(k, v)]
+toListT (Branch pref ma prefMap) = case ma of
+    Just a  -> (:) (pref, a)
+    Nothing -> id
+    $ map (\(k, v) -> (pref <> k, v)) $ toList prefMap
+
+-- | Converts 'PrefixMap' to the list of pairs.
+toList :: PrefixMap a -> [(Key, a)]
+toList tree = concatMap (\(p, tr) -> map (addPiece p) $ toListT tr) $ HashMap.toList tree
```

Comment:
You could write this function in pointfree by removing `tree` variable name. And also you can rewrite

```haskell
map f $ x
```

to

```haskell
f <$> x
```